jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_class.cpp

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting annotations
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <java-testing-utils/load_java_class.h>
+#include <java_bytecode/java_bytecode_convert_class.h>
+#include <java_bytecode/java_bytecode_parse_tree.h>
+#include <java_bytecode/java_types.h>
+#include <testing-utils/catch.hpp>
+
+SCENARIO(
+  "java_bytecode_parse_class",
+  "[core][java_bytecode][java_bytecode_parser]")
+{
+  GIVEN("Some class with a class hierarchy")
+  {
+    const symbol_tablet &new_symbol_table =
+      load_java_class("ChildClass", "./java_bytecode/java_bytecode_parser");
+    WHEN("Parsing the class file structure")
+    {
+      THEN("We should be able to get the super classes of the hierarchy")
+      {
+        const symbolt &class_symbol1 =
+          new_symbol_table.lookup_ref("java::ChildClass");
+        const java_class_typet java_class1 =
+          to_java_class_type(class_symbol1.type);
+        REQUIRE(java_class1.get_super_class() == "ParentClass");
```

Comment:
FYI: You can have multiple `THEN` and `WHEN` sections. That way you'll end up with more smaller tests. The benefit (apart from better variable names and shorter test body) is if one of the asserts below fails, all others will still be checked independently.